packages/wasm-parser/src/decoder.js

```diff
@@ -1752,12 +1752,17 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
           t.sectionMetadata("custom", startOffset, sectionSizeInBytesNode)
         ];
 
+        const offsetBeforeReadUTF8 = offset;
         const sectionName = readUTF8String();
+        const bytesEatenByReadUTF8 = offset - offsetBeforeReadUTF8;
         eatBytes(sectionName.nextIndex);
 
         dump([], `section name (${sectionName.value})`);
 
-        const remainingBytes = sectionSizeInBytes - sectionName.nextIndex;
+        // B/c readUTF8String includes an eatBytes call, we have to account
+        // for bytes eaten by readUTF8String itself
```

Comment:
That's a good catch! My previous comment (that you can ignore then) assumed that. That's confusing.